mountpoint-s3/src/cli.rs

```diff
@@ -54,7 +53,7 @@ Arguments:
     after_help = if cfg!(feature = "fstab") {FSTAB_DOCS} else {""},
 )]
 pub struct CliArgs {
-    #[clap(help = "Name of bucket to mount", value_parser = parse_bucket_name)]
+    #[clap(help = "Name of bucket, or an S3 URI, to mount")]
     pub bucket_name: String,
```

Comment:
As per @dannycjones's comment, maybe this should be 
```suggestion
    pub bucket_name: S3Path,
```
and we should keep value parser here? So we don't have to change each method to return a `Result`?

Reply:
We can't change this to `S3Path` directly here, as it needs to take the prefix argument into account